src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1028 @@
+        r"embed_tokens.weight",
+    ]
+
+    def __init__(self, config: GPTSANJapaneseConfig):
+        super().__init__(config)
+        self.embed_tokens = nn.Embedding(config.vocab_size, config.d_model)
+        self.position_embeddings = nn.Embedding(config.num_contexts, config.d_model)
+        self.token_bias = nn.Parameter(torch.zeros([config.vocab_size]))
+        self.logits = nn.Linear(config.d_model, config.d_model, bias=True)
+        self.logact = self.act = ACT2FN["swish"]
+        self.config = copy.deepcopy(config)
+
+        self.blocks = torch.nn.ModuleList([])
+        for _ in range(config.num_switch_layers):
+            self.blocks.append(GPTSANJapaneseBlock(config))
+        for _ in range(config.num_ext_layers):
+            self.blocks.append(GPTSANJapaneseBlock(config, ext_layer=True))
+
+        if config.num_ext_layers > 0:
+            self.extra_position_embeddings = nn.Embedding(config.num_contexts, config.d_model)
+
+        if config.d_spout:
+            spouts = []
+            for _ in range(8):
+                spouts.append(nn.Linear(config.d_spout, config.d_spout, bias=False))
+                spouts.append(nn.Tanh())
+            spouts.append(nn.Linear(config.d_spout, config.num_layers * 2 * config.d_model, bias=False))
+            self.spout = nn.Sequential(*spouts)
+
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.embed_tokens
+
+    def set_input_embeddings(self, new_embeddings):
+        self.embed_tokens = new_embeddings
+
+    @add_start_docstrings_to_model_forward(GPTSAN_JAPANESE_INPUTS_DOCSTRING)
+    def forward(
+        self,
+        input_ids: Optional[torch.LongTensor] = None,
+        num_precontext: Optional[torch.LongTensor] = None,
+        spout: Optional[torch.FloatTensor] = None,
+        past_key_values: Optional[Tuple[Tuple[torch.FloatTensor]]] = None,
+        labels: Optional[torch.LongTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        use_cache: Optional[bool] = False,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        decoder_inputs_embeds: Optional[torch.FloatTensor] = None,
+        output_attentions: Optional[bool] = False,
+        output_hidden_states: Optional[bool] = False,
+        return_dict: Optional[bool] = False,
+        output_router_logits: Optional[bool] = False,
+    ) -> Union[Tuple[torch.FloatTensor], ModelOutput]:
+        r"""
+        Returns:
+            `ModelOutput` or `namedtuple` if `return_dict` returns ModelOutput insted of namedtuple
+
+        Example:
+
+        ```python
+        from transformers import AutoModel, AutoTokenizer
+
+        model = AutoModel.from_pretrained("Tanrei/GPTSAN-japanese")
+        tokenizer = AutoTokenizer.from_pretrained("Tanrei/GPTSAN-japanese")
+        x_tok = tokenizer.encode("武田信玄は、", return_tensors="pt")
+        model = model.cuda()
+        c = model.generate(x_tok.cuda(), max_new_tokens=50, random_seed=63)
```

Comment:
Not really that this should work. THe tokenizer returns a dictionary of {"inpud_ids":..., "attention_maks":,...} and even `token_type_ids` if it follows bert's 2 sentence input. Make sur to add GPTSANJapanese to the `documentation_text.txt` file and run the doctests

Reply:
Also avoid single letter variables

Reply:
@ArthurZucker 
I'm confused. This tokenizer is based on GPT-NeoX-Japanese,
```
>>> tokenizer = AutoTokenizer.from_pretrained("abeja/gpt-neox-japanese-2.7b")
>>> tokenizer.encode("武田信玄は、", return_tensors="pt")
tensor([[ 8640, 25948, 26638, 31678]])
>>> tokenizer.encode("武田信玄は、", return_tensors="pt").input_ids
Traceback (most recent call last):
   File "<stdin>", line 1, in <module>
AttributeError: 'Tensor' object has no attribute 'input_ids'
```
It has the same behavior

Reply:
I see. I was wrong then! 🙇🏻‍♂️ usually all of our tokenizer have the “input_ids” attribute (as well as “attention_mask”) which is why I was confused. 
only comment then is not to use single letter variable (c -> generated_ids) and still make sure the doc tests pass